### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by carlosbollero**
### Summary

Add multiple step registration:

* Adjust registration screen to show multiple steps.
* Add formik for form state management.
* Add basic validations as an example.

### Screenshots

|| Screenshot |
|:--:|:--:|
| Steps | registration |
| Validation | image |
| Validation | image |

### CODE REVIEW

**Review by mcallegari10 — APPROVED** [review]

File: generators/app/templates/src/app/screens/Registration/components/FormStep/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+.input {
+  border: solid 1px;
+  border-radius: 5px;
+  font-size: 14px;
+  height: 30px;
+  padding: 0 5px;
+  width: 100%;
+  min-width: 250px;
+}
```
Comment (tomiir) [inline generators/app/templates/src/app/screens/Registration/components/FormStep/styles.module.scss]: missing EOF

File: generators/app/templates/src/app/screens/Registration/index.tsx [hunk 2]
```diff
@@ -1,89 +1,86 @@
...
+    }
+  });
+
+  const handleFormSubmit = (values: UserData) => {
+    signUpRequest(values);
+  };
+
+  const formik = useFormik({ initialValues, onSubmit: handleFormSubmit, validate });
+
+  const { handleChange, handleSubmit, values, errors, isValid } = formik;
```
Comment (mcallegari10) [inline generators/app/templates/src/app/screens/Registration/index.tsx]: Can't you destructure this variables in the first line? Like:

```suggestion
  const { handleChange, handleSubmit, values, errors, isValid } = useFormik({ initialValues, onSubmit: handleFormSubmit, validate });
```

File: generators/app/templates/src/app/screens/Registration/constants.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  ]
+};
+
+export const steps: any[] = [personalData, address, credentials];
+
+export const initialValues = steps.reduce(
+  (acum, step) => ({
+    ...acum,
+    ...step.fields.reduce((acu: any, field: any) => ({ ...acu, [field.name]: field.initialValue }), {})
```
Comment (mcallegari10) [inline generators/app/templates/src/app/screens/Registration/constants.ts]: Add a `TODO`, I think we can type this more strongly, like declaring a `field` interface with all the posible validations

**Review by mcallegari10 — APPROVED** [review]

File: generators/app/templates/src/app/screens/Registration/index.tsx [hunk 4]
```diff
@@ -1,89 +1,89 @@
...
+      setStep(currentStep + 1);
+    }
+  };
 
-  const handleInputChange = useCallback((event: React.FormEvent<Element>) => {
-    // TODO: Implement method
-  }, []);
+  const handlePrevStep = () => {
+    if (currentStep > 0) {
+      setStep(currentStep - 1);
```
Comment (FrankIglesias) [inline generators/app/templates/src/app/screens/Registration/index.tsx]:
```suggestion
      setStep(Math.max(0,currentStep));
```

File: generators/app/templates/src/app/screens/Registration/index.tsx [hunk 5]
```diff
@@ -1,89 +1,89 @@
...
+  const [currentStep, setStep] = useState(0);
 
-function RegistrationContainer() {
-  /* eslint-disable @typescript-eslint/no-unused-vars */
-  const handleSubmit = useCallback((event: React.FormEvent<Element>) => {
-    // TODO: Implement
-  }, []);
+  const handleNextStep = () => {
+    if (currentStep < steps.length - 1) {
+      setStep(currentStep + 1);
```
Comment (FrankIglesias) [inline generators/app/templates/src/app/screens/Registration/index.tsx]:
```suggestion
      setStep(Math.min(currentStep + 1, steps.length-1));
```

File: generators/app/templates/src/app/screens/Registration/index.tsx [hunk 6]
```diff
@@ -1,89 +1,89 @@
...
+        className="m-bottom-5"
+        id={id}
+        name={name}
+        fields={fields}
+        values={values}
+        handleChange={handleChange}
+        errors={errors}
+      />
+      <div className={`row space-between full-width ${styles.formButtons}`}>
+        <button className={styles.button} type="button" onClick={handlePrevStep} disabled={currentStep === 0}>
```
Comment (FrankIglesias) [inline generators/app/templates/src/app/screens/Registration/index.tsx]:
```suggestion
        <button className={styles.button} type="button" onClick={handlePrevStep} disabled={!currentStep}>
```

File: generators/app/templates/src/app/screens/Registration/validations.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import { required, email } from '~utils/inputValidations';
+
+interface Validations {
+  [key: string]: ((val: any) => any)[];
+}
+
+const validations: Validations = {
+  firstName: [required('Requerido')],
+  email: [required('Requerido'), email('Email incorrecto')]
+};
```
Comment (FrankIglesias) [inline generators/app/templates/src/app/screens/Registration/validations.ts]: missing i18n

File: generators/app/templates/src/app/screens/Registration/validations.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+export const validate = (values: any) => {
+  const errors: { [key: string]: string } = {};
+  Object.keys(values).forEach((key: string) => {
+    validations[key]?.forEach((validation: (arg: string) => string) => {
+      const valResult = validation(values[key]);
+      if (valResult) {
+        errors[key] = valResult;
+      }
```
Comment (FrankIglesias) [inline generators/app/templates/src/app/screens/Registration/validations.ts]: This can be replaced  with a filter

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

**Comment by pabloferro** [comment]
This has too many conflicts and is outdated. We should revisit this feature in another PR.

**Pull request closed without merging**